Ijaegle nonlinear

igjaegle opened the pull request:
Add:
1/ version 3 of non-linear correction
2/ Updates eta to 2g and 6g skimmer

gluex commented:
Test status for this pull request: FAILURE
 
 Failures: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/failures.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-failures.txt)
 Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

gluex commented:
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

gluex commented:
Test status for this pull request: FAILURE
 
 Failures: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/failures.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-failures.txt)
 Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

gluex commented:
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

sdobbs commented on src/libraries/FCAL/DFCALShower_factory.cc:
```diff
@@ -203,6 +207,21 @@ jerror_t DFCALShower_factory::brun(JEventLoop *loop, int32_t runnumber)
 	}
       }
     }
+    loop->GetCalib("FCAL/nonlinear", nonlinear);
```
I am not crazy about the name of this table.  Can it be more descriptive, like "FCAL/nonlinear_correction", or something like that?

igjaegle commented:
Sure, no problem, I will change to FCAL/nonlinear_correction

tks ig.

On Thu, Jan 25, 2024 at 10:03 AM Sean Dobbs ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/libraries/FCAL/DFCALShower_factory.cc
> <https://github.com/JeffersonLab/halld_recon/pull/779#discussion_r1466506596>
> :
>
> > @@ -203,6 +207,21 @@ jerror_t DFCALShower_factory::brun(JEventLoop *loop, int32_t runnumber)
>  	}
>        }
>      }
> +    loop->GetCalib("FCAL/nonlinear", nonlinear);
>
> I am not crazy about the name of this table. Can it be more descriptive,
> like "FCAL/nonlinear_correction", or something like that?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JeffersonLab/halld_recon/pull/779#pullrequestreview-1843955387>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFL3GRXEFQ32CFXMH6SHGVLYQJX2HAVCNFSM6AAAAABCJA3DJWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNBTHE2TKMZYG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

sdobbs commented:
I think this generally looks fine, but since we're working with this section of the code, can we add a CCDB table that specifies which non-linear correction to use?  It can just contain a single integer, which would select which nonlinear correction to use.

sdobbs commented:
I'm not feeling very creative with names this morning, maybe "FCAL/nonlinear_correction_form"?
Other suggestions are welcome.

gluex commented:
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

gluex commented:
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

igjaegle commented:
There are some inconsistencies that I am correcting. I should push a new version soon.

gluex commented:
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

sdobbs commented:
Thanks a lot!  looks good except for two things:
* I think that you accidentally put in a file `src/libraries/DFCALShower_factory.cc` (outside of libraries/FCAL)
* Can you also add in values for the default variation for the other run periods for /FCAL/nonlinear_correction_type?  It looks like these values are mostly in a separate variation.

igjaegle commented:
Thank you, Sean. 
1/ I removed the misplaced file and check that the correct file in libraries/FCAL
2/ I added to DB the /FCAL/nonlinear_correction_type for all calibrated run periods.

gluex commented:
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

sdobbs reviewed (approved):
looks good, thanks!
